MatrixSDK/Data/MXRoom.m

```diff
@@ -2072,6 +2072,22 @@ - (MXHTTPOperation*)redactEvent:(NSString*)eventId
     return [mxSession.matrixRestClient redactEvent:eventId inRoom:self.roomId reason:reason success:success failure:failure];
 }
 
+- (MXHTTPOperation*)redactEvent:(NSString*)eventId
+                  withRelations:(NSArray<NSString *>*)relations
+                         reason:(NSString*)reason
+                        success:(void (^)(void))success
+                        failure:(void (^)(NSError *error))failure
+{
+    BOOL isFeatureSupported = (mxSession.store.supportedMatrixVersions.supportsRedactionWithRelations || mxSession.store.supportedMatrixVersions.supportsRedactionWithRelationsUnstable);
+    BOOL isFeatureStable = mxSession.store.supportedMatrixVersions.supportsRedactionWithRelations;
+    if (isFeatureSupported) {
+        return [mxSession.matrixRestClient redactEvent:eventId withRelations:relations inRoom:self.roomId reason:reason txnId:nil featureIsStable:isFeatureStable success:success failure:failure];
+    } else {
+        MXLogDebug(@"[MXRoom] redaction with relations is not supported (MSC3912)");
+        return [self redactEvent:eventId reason:reason success:success failure:failure];
```

Comment:
Shouldn't we mark this method as deprecated since it is called by `- (MXHTTPOperation*)redactEvent:(NSString*)eventId withRelations:(NSArray<NSString *>*)relations reason:(NSString*)reason success:(void (^)(void))success failure:(void (^)(NSError *error))failure` if the server doesn't support this new MSC?

Reply:
This is too early to mark it as deprecated. This would be the case eventually when MSC3912 will be stable

Reply:
After thinking we think you're right. We should not use the previous endpoint anymore.
So we will call the most [recent endpoint](https://spec.matrix.org/unstable/client-server-api/#put_matrixclientv3roomsroomidredacteventidtxnid) defined for the redaction by ignoring the relations

@nimau will update his PR

Reply:
Thx @gileluard , I checked this point with @nimau. Finally he will call here the new implemented redact function (based on the actual [redact endpoint](https://spec.matrix.org/latest/client-server-api/#put_matrixclientv3roomsroomidredacteventidtxnid)) by ignoring the relations

Reply:
@gileluard: I fixed it in the last commit. Now the fallback (if the MSC3912 is not supported) will be on the new endpoint but without passing any relation types.